site/docs/v1/tech/guides/breached-password-detection.adoc

```diff
@@ -0,0 +1,274 @@
+  "fieldErrors": {
+    "user.password": [
+      {
+        "code": "[breachedCommonPassword]user.password",
+        "message": "The [user.password] property value has been breached and may not be used, please select a different password."
+      }
+    ]
+  }
+}
+----
+
+If you enable breach detection on login and don't require a password change, you'll receive a `200` HTTP status code on successful login and the normal JSON response. Please see the link:../apis/login[Login API] documentation for more.
+
+If you require a password change, you'll receive a `203` HTTP status code should the user's password have been compromised. You'll also receive JSON explaining why the password change is required:
+
+.JSON response when attempted user created with a breached password
+[source,json]
+----
+{
+  "changePasswordId": "yo0FiR_y7Rrtrk2vKe_F93PQ-nWljfGGWexgNSbHfXQ",
+  "changePasswordReason": "Breached"
+}
+----
+
+You can then use the link:../apis/users#change-a-users-password[Change a User's Password API] to, well, change the user's password.
+
+== How It Works
+
+Here's a high level diagram of the FusionAuth breached password detection service:
+
+image::guides/breached-password-detection/breached-password-detection-diagram.png[Diagram of the breached password detection system.,width=700,role=box-shadow margin-right]
+
+New breached passwords are ingested and processed by FusionAuth on an ongoing basis. An additional encryption key is used to securely communicate with the FusionAuth Reactor service using a strong AES cipher. 
+
+You can rotate this key by navigating to the [breadcrumb]#Reactor# tab and clicking the *Regenerate Key* button. When you do so, you'll be prompted to confirm this decision:
+
+image::guides/breached-password-detection/regenerating-the-key.png[Regenerating the AES key.,width=700,role=box-shadow margin-right]
+
+== Limitations
+
+Breached password detection in FusionAuth has a few limitations of which you should be aware.
+
+As you might expect, it can't check passwords controlled by other systems, such as when you use an link:../identity-providers/[external identity provider].
+
+All other password rules, as defined in the tenant configuration, apply when passwords are changed. For example, if you set a minimum length of 25 characters for passwords in addition to enabling breached password detection, both conditions must be satisfied for a password to be accepted. The user will have to select a password that is at least 25 characters long and not known to have been compromised.
+
+Breached password detection is configured at the tenant level and applies to all applications within that tenant. For example, if you enable detection for the `Default` tenant, where the FusionAuth application resides, administrative users with access to that application will have their passwords checked.
+
+Due to the salting and computational intensity of checking passwords, there's no way to examine passwords in an offline or batch fashion.
```

Comment:
Wondering if should just omit this since it may raise more questions.

Reply:
Sure.